internal/agent/config_test.go

```diff
@@ -0,0 +1,37 @@
+package agent
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+var yamlConfig = `management-endpoint: https://management.endpoint
+enrollment-endpoint: https://management.endpoint
+enrollment-ui-endpoint: https://ui.management.endpoint
+spec-fetch-interval: 0m10s
+status-update-interval: 0m10s`
```

Comment:
I would also check the data-dir , and the expected file paths to be in the data dir to be generated accordingly

Reply:
the data dir (/var/lib/flightctl) and config dir should be part of the rpm spec IIUC. The cert directory inside configDir for example, could be created by the agent.

Reply:
OK we should explicitly define what directories the config must ensure

```
EnrollmentCertFile
EnrollmentKeyFile 
GeneratedCert
Cacert
Key
```

Reply:
EnrollmentCertFile, EnfollmentKeyFile and Cacert are mandatory for the agent's bootstrap. Key and GeneratedCert are created by the agent at runtime.

Reply:
what about TPMPath

Reply:
Not mandatory so far, but I defer to @fzdarsky

Reply:
> EnrollmentCertFile, EnfollmentKeyFile and Cacert are mandatory for the agent's bootstrap. Key and GeneratedCert are created by the agent at runtime.

right but the directories must exist do we fail if they don't or do we create them. lets solve this for good :)

Reply:
> EnrollmentCertFile, EnfollmentKeyFile and Cacert are mandatory for the agent's bootstrap. Key and GeneratedCert are created by the agent at runtime.

But these are not directories, but assets. Directories are only "ConfigDir and DataDir". right?

Reply:
IE if I define Key: /some/strange/path is that an error if it doesn't exist or should it always be created. I assume error is worse in this case but then also what is an invalid path?